services/web/server/src/simcore_service_webserver/folders/_folders_db.py

```diff
@@ -0,0 +1,223 @@
+""" Database API
+
+    - Adds a layer to the postgres API with a focus on the projects comments
+
+"""
+
+import logging
+from typing import cast
+
+from aiohttp import web
+from models_library.folders import FolderDB, FolderID
+from models_library.products import ProductName
+from models_library.rest_ordering import OrderBy, OrderDirection
+from models_library.users import GroupID, UserID
+from models_library.workspaces import WorkspaceID
+from pydantic import NonNegativeInt, parse_obj_as
+from simcore_postgres_database.models.folders_v2 import folders_v2
+from sqlalchemy import func
+from sqlalchemy.sql import asc, desc, select
+
+from ..db.plugin import get_database_engine
+from .errors import FolderAccessForbiddenError, FolderNotFoundError
+
+_logger = logging.getLogger(__name__)
+
+
+_SELECTION_ARGS = (
+    folders_v2.c.folder_id,
+    folders_v2.c.name,
+    folders_v2.c.parent_folder_id,
+    folders_v2.c.created_by_gid,
+    folders_v2.c.created,
+    folders_v2.c.modified,
+    folders_v2.c.user_id,
+    folders_v2.c.workspace_id,
+)
+
+
+async def create_folder(
+    app: web.Application,
+    product_name: ProductName,
+    created_by_gid: GroupID,
+    folder_name: str,
+    parent_folder_id: FolderID | None,
+    user_id: UserID | None,
+    workspace_id: WorkspaceID | None,
+) -> FolderDB:
+    assert not (
+        user_id is not None and workspace_id is not None
+    ), "Both user_id and workspace_id cannot be provided at the same time. Please provide only one."
+
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(
+            folders_v2.insert()
+            .values(
+                name=folder_name,
+                parent_folder_id=parent_folder_id,
+                product_name=product_name,
+                user_id=user_id,
+                workspace_id=workspace_id,
+                created_by_gid=created_by_gid,
+                created=func.now(),
+                modified=func.now(),
+            )
+            .returning(*_SELECTION_ARGS)
+        )
+        row = await result.first()
+        return parse_obj_as(FolderDB, row)
+
+
+async def list_folders(
```

Comment:
I would remove all these folder/folders of the function names. we are already in folders after all